ingestion/src/metadata/profiler/processor/sampler/sqlalchemy/sampler.py

```diff
@@ -63,43 +63,45 @@ class SQASampler(SamplerInterface):
     run the query in the whole table.
     """
 
-    def get_sample_percentage_cte(self):
-        return (
-            self.client.query(
-                self.table,
-                (ModuloFn(RandomNumFn(), 100)).label(RANDOM_LABEL),
+    def random_sampler_where_clause_(self, query):
+        """To override where clause to random sampler"""
+        return query
```

Comment:
what I mean is convert this part from the `get_sample_query`

```
self.client.query(
                     self.table,
                     (ModuloFn(RandomNumFn(), 100)).label(RANDOM_LABEL),
                 )
```
into something like

```
self._base_sample_query((ModuloFn(RandomNumFn(), 100)).label(RANDOM_LABEL))
```

where `_base_sample_query ` is something like that

```
def _base_sample_query(random_fn):
    return  self.client.query(
                     self.table,
                     random_fn,
                 )
```

Then in Trino override `_base_sample_query ` with the where logic. Since this query is the base one that retrieve the data it should be enough to just do the filtering at this level.